### ISSUE TITLE
Services loosing icon

### ISSUE THREAD

**Issue by RJVB** [ferdium/ferdium-app#1252]
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/ferdium/ferdium-app/blob/HEAD/CODE_OF_CONDUCT.md) that Ferdium Recipes adheres to.
- [X] I have searched the [issue tracker](https://github.com/ferdium/ferdium-app/issues?q=is%3Aopen+is%3Aissue+label%3A%22recipe+%3Abookmark_tabs%3A%22+label%3A%22bug+%3Abeetle%3A%22+) for a bug report that matches the one I want to file, without success.

### Ferdium Version

6.3.0

### Recipe Name

Messenger

### Recipe Version

6.3.0?

### Steps to reproduce

I've started noticing since Ferdium 6.2.7 that the Messenger service loses its icon. I've been using my profile since `Ferdi` became unusable so I cannot give a clear step 1.

I can give this series of steps:
1) launch Ferdium
2) if the Messenger service doesn't have its icon, copy `icon.svg` from the old `~/.config/Ferdi/recipes/messenger` folder
3) warm-restart or quit/restart Ferdium
4) the icon will be there
5) quit Ferdium
6) start Ferdium the following day after suspending the computer for the night
7) icon gone again, *including* the file in `~/.config/Ferdium/recipes/messenger`

### Expected Behavior

Service should show its default icon unless I tell it to use another one (which I didn't AS FAR AS I CAN REMEMBER).

### Actual Behavior

Service doesn't keep its icon

### Additional Information

Is there a way to be certain that the service is configured to use its default icon, and not some file that doesn't exist, no matter how unlikely that is (given how it will use the `icon.svg` file in the appropriate location)?

**Comment by RJVB** [comment]
I've change the title of this issue as it is no longer only the Messenger service. A number of my own custom services (for discussion forums) exhibit the same behaviour.

The `icon.svg` files no longer seem to disappear from the recipe directories though. I could just restore the icon display on the Messenger service by deleting the custom icon I had apparently configured, but that trick doesn't work for my custom services.

This is with 6.4.0 .

**Comment by Arthur-Huan** [comment]
This has happened to me before. I was using a custom icon for Messenger before updating the recipe on here. After the change, the custom icon was no longer needed, and I deleted the icon in `~/.config/Ferdium/icons/`. However, the service was still trying to get the custom icon from that folder instead of the recipes.

Could you check that when you go and Edit the service, the “Custom icon” is unset? It should say “Drop your image, or click here”, otherwise Ferdium thinks you're using a custom icon.

**Comment by RJVB** [comment]
>Could you check that when you go and Edit the service, the “Custom icon” is unset? 

Well, it was again thinking that I had a custom icon set, so I deleted it. Again... I have to do that every I start Ferdium.

Or use the View/Reload Ferdium action.

I don't have a ~/.config/Ferdium/icons directory; maybe that's a personal place where you keep your custom icons?

I tried deleting ~/.config/Ferdium/recipes/messenger/icon.svg but then the service remains without an icon even after I delete the supposed custom icon.

**Comment by Arthur-Huan** [comment]
`~/.config/Ferdium/recipes/messenger/icon.svg` is not what I'm talking about when I say “custom icon”. That's the recipe icon. Even if you delete or change it, to Ferdium it's considered the default recipe icon, not a custom icon.

If you don't have custom icons set, the settings should look like this: 

![image](https://github.com/ferdium/ferdium-app/assets/106561822/5f6cf95b-991b-484a-b410-201e47a60950)

If it doesn't look like that, that means you have set a custom icon. Ferdium will ignore the icon in the recipes folder, so deleting or changing that file will do nothing.

Any custom icon you give Ferdium will automatically be saved to `~/.config/Ferdium/icons`. It will ignore the recipe icon and go to this directory to get your custom icon. Since you don't have the directory, it won't find an icon and will just display nothing.

**Comment by RJVB** [comment]
Be that as it may, I shouldn't have to delete that supposed custom icon I set every time I restart/reload Ferdium!

**Comment by Arthur-Huan** [comment]
The expected behavior is that you just have to click that little square in the settings to remove it, click “Save service”, and you should no longer need to make any more changes. That's what happens when I try. Is this not the case for you?

**Comment by RJVB** [comment]
> Is this not the case for you?

As I tried to indicate, that's what happens, but only until I quit or reload Ferdium. After that I have to do it again.

**Comment by RJVB** [comment]
>Any custom icon you give Ferdium will automatically be saved to `~/.config/Ferdium/icons`.

I created the directory thinking maybe that'll help, but nothing gets saved in there.

**Comment by SpecialAro** [comment]
Hi @RJVB

I'm not entirely sure if this issue is the same as https://github.com/ferdium/ferdium-app/issues/1373 but, if it is, I just raised a PR to fix it on https://github.com/ferdium/ferdium-server/pull/78. Once this is merged and the docker image is updated on the production environment (on Ferdium Server) then this issue will be (possibly) fixed.

**Comment by RJVB** [comment]
Keeping fingers crossed then!

**Comment by SpecialAro** [comment]
> Keeping fingers crossed then!

To each service that have a missing icon, right click, edit, click on the delete icon button (like if you were updating a custom Icon) and save the service. Repeat until all services are fixed.

This should work 😄 (at least it fixed my missing icon for Steam)

**Comment by NocturnalNick** [comment]
> > Keeping fingers crossed then!
> 
> To each service that have a missing icon, right click, edit, click on the delete icon button (like if you were updating a custom Icon) and save the service. Repeat until all services are fixed.
> 
> This should work 😄 (at least it fixed my missing icon for Steam)

All it did for me was replace the previous custom icon with the default wireframe sphere icon, so now have to go through and recreate.

**Comment by SpecialAro** [comment]
> > > Keeping fingers crossed then!
> > 
> > To each service that have a missing icon, right click, edit, click on the delete icon button (like if you were updating a custom Icon) and save the service. Repeat until all services are fixed.
> > 
> > This should work 😄 (at least it fixed my missing icon for Steam)
> 
> All it did for me was replace the previous custom icon with the default wireframe sphere icon, so now have to go through and recreate.

Yes, sorry about that... the uploaded icons got lost because we weren't mounting the path on the docker container on the server side in order to save them whenever we restarted the container. Now I've manually set a mountpoint so it should save the icons properly.

**Comment by NocturnalNick** [comment]
> > > > Keeping fingers crossed then!
> > > 
> > > 
> > > To each service that have a missing icon, right click, edit, click on the delete icon button (like if you were updating a custom Icon) and save the service. Repeat until all services are fixed.
> > > This should work 😄 (at least it fixed my missing icon for Steam)
> > 
> > 
> > All it did for me was replace the previous custom icon with the default wireframe sphere icon, so now have to go through and recreate.
> 
> Yes, sorry about that... the uploaded icons got lost because we weren't mounting the path on the docker container on the server side in order to save them whenever we restarted the container. Now I've manually set a mountpoint so it should save the icons properly.

And the award for best unpaid customer service goes to...

Do I need to re-set them now that you've made that change so they're re-set on the server side?

**Comment by SpecialAro** [comment]
Thank you for your kindness ☺️

Yes 😁 you should re-set them now. But be aware that the fix I did on the server side is only temporary - soon we will merge a huge PR that will update a big dependency we rely on (meaning that possibly once we merge you might loose your icons again - hopefully not if I've code everything right).

This is just a heads-up for you not to get mad on Ferdium 🙈

**Comment by NocturnalNick** [comment]
> Thank you for your kindness ☺️
> 
> Yes 😁 you should re-set them now. But be aware that the fix I did on the server side is only temporary - soon we will merge a huge PR that will update a big dependency we rely on (meaning that possibly once we merge you might loose your icons again - hopefully not if I've code everything right).
> 
> This is just a heads-up for you not to get mad on Ferdium 🙈

Great advice. I'll probably not both resetting them just yet then, and will do so next time they're lost (I've kept the image files this time).

Thanks again.

**Comment by SpecialAro** [comment]
> > Thank you for your kindness ☺️
> > 
> > Yes 😁 you should re-set them now. But be aware that the fix I did on the server side is only temporary - soon we will merge a huge PR that will update a big dependency we rely on (meaning that possibly once we merge you might loose your icons again - hopefully not if I've code everything right).
> > 
> > This is just a heads-up for you not to get mad on Ferdium 🙈
> 
> Great advice. I'll probably not both resetting them just yet then, and will do so next time they're lost (I've kept the image files this time).
> 
> Thanks again.

Hey!

It is not a certain that they will get lost again, probably they won't because I backported the code. 😁

**Comment by SpecialAro** [comment]
Closing this issue as it is fixed.
